src/views/ProductView.vue

```diff
@@ -744,173 +750,271 @@ function goToWishlists() {
+        return count != undefined && count > 0
+    }
+
+    return (
+        productItemsCountOfItemsInStock.value != undefined &&
+        productItemsCountOfItemsInStock.value > 0
+    )
+})
 
 /**
- * Asynchronously computes the inventory status of product items.
+ * Gets the number of product items in stock
+ * by "querying" the product variant object.
  */
-const getInventoryStatusOfProductItemsQuery = asyncComputed(
+function getProductItemsCountOfItemsInStockForBuyer(): number | undefined {
+    if (productVariantInfoRelevantToBuyer.value?.productItems != undefined) {
+        return productVariantInfoRelevantToBuyer.value.productItems.totalCount
+    } else {
+        return undefined
+    }
+}
+
+/**
+ * A ref to trigger the evaluation of the inventory status of the product variant:
+ * For each product item status, how many product items have that status?
+ */
+const trigger = ref(0)
+
+/**
+ * Asynchronously computed property for the number of product items in stock.
+ * @returns The number of product items in stock or undefined if no data is available.
+ */
+const productItemsCountOfItemsInStock = asyncComputed(
     async () => {
-        triggerGetInventoryStatusOfProductItemsQuery.value
+        trigger.value
 
-        if (productVariantId.value !== undefined) {
-            return client.getInventoryStatusOfProductItems({
-                productVariantId: productVariantId.value,
-            })
+        if (!activeUserRoleIsEitherAdminOrEmployee.value) {
+            return getProductItemsCountOfItemsInStockForBuyer()
         }
 
-        return null
+        const queryResponse = await getProductItemsCountOfInventoryStatus(
+            ProductItemStatus.InStorage
+        )
+
+        if (queryResponse != undefined) {
+            return queryResponse.productItems.totalCount
+        } else {
+            return undefined
+        }
     },
-    null,
+    undefined,
     {
         onError: (e) =>
             pushErrorNotification(
-                errorMessages.getInventoryStatusOfProductItems,
+                errorMessages.getProductItemsCountOfInventoryStatus,
                 e
             ),
         shallow: false,
     }
 )
 
 /**
- * Computed property for the inventory status of product items.
+ * Asynchronously computed property for the number of reserved product items.
+ * @returns The number of reserved product items or undefined if no data is available.
  */
-const inventoryStatusOfProductItems = computed(() => {
-    if (getInventoryStatusOfProductItemsQuery.value !== null) {
-        const nodes =
-            getInventoryStatusOfProductItemsQuery.value
-                .productItemsByProductVariant.nodes
-        if (nodes !== null && nodes !== undefined) {
-            return nodes
+const productItemsCountOfReservedItems = asyncComputed(
+    async () => {
+        trigger.value
+
+        const queryResponse = await getProductItemsCountOfInventoryStatus(
```

Comment:
How about putting the part which is duplicated in a function?

Reply:
67ad16477c1e8d4e7168d03ba2e51415e30ba075